### ISSUE TITLE
Test 4 fails in "Get JSON with the JavaScript XMLHttpRequest Method"

### ISSUE THREAD

**Issue by stephanmax** [freeCodeCamp/freeCodeCamp#13999]
Challenge [get-json-with-the-javascript-xmlhttprequest-method](https://beta.freecodecamp.com/en/challenges/json-apis-and-ajax/get-json-with-the-javascript-xmlhttprequest-method) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</code>.

Solution below fails for same reason as reported in #13571 (extra spacing in function assignment to `req.onload`). Solution proposed in said ticket sounds solid and I am happy to file a PR for this one.

```html


<script>
  document.addEventListener('DOMContentLoaded',function(){
    document.getElementById('getMessage').onclick=function(){
      // Add your code below this line
      var req = new XMLHttpRequest();
      req.open("GET", '/json/cats.json', true);
      req.send();
      req.onload = function(){
        var json = JSON.parse(req.responseText);
        document.getElementsByClassName('message')[0].innerHTML=JSON.stringify(json);
      };
      
      // Add your code above this line
    };
  });
</script>
<style>
  body {
    text-align: center;
    font-family: "Helvetica", sans-serif;
  }
  h1 {
    font-size: 2em;
    font-weight: bold;
  }
  .box {
    border-radius: 5px;
    background-color: #eee;
    padding: 20px 5px;
  }
  button {
    color: white;
    background-color: #4791d0;
    border-radius: 5px;
    border: 1px solid #4791d0;
    padding: 5px 10px 8px 10px;
  }
  button:hover {
    background-color: #0F5897;
    border: 1px solid #0F5897;
  }
</style>
<h1>Cat Photo Finder</h1> 
<p class="message box">
  The message will go here
</p>
<p>
  <button id="getMessage">
    Get Message
  </button>
</p>

```

**Comment by QuincyLarson** [comment]
@stephanmax yes - I agree. We would welcome a PR from you that addresses this 👍

**Comment by QuincyLarson** [comment]
@stephanmax have you had a chance to take a look at this yet?

**Comment by systimotic** [comment]
## First-timers only
Check out [CONTRIBUTING.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/CONTRIBUTING.md) to get your local development environment set up.

This test is on this line:
https://github.com/freeCodeCamp/freeCodeCamp/blob/9f7ca2bcf5f4af3de74dbd0ec375e7c570d47f32/seed/challenges/04-data-visualization/json-apis-and-ajax.json#L235
The regular expression will need to allow spaces between `onload`, `=`, and `function`. To do this, add `\\s*` in places where spaces are allowed but not required.

If you need help, please go to our [contributors chatroom](https://gitter.im/FreeCodeCamp/Contributors). We're always happy to answer any questions!

Happy coding!

**Comment by Vaidic** [comment]
This is my first time getting started with open source software. Can I work on this?

**Comment by systimotic** [comment]
@Vaidic For sure! Remember, if you need help, you can find us in the contributors chatroom linked above.

**Comment by andrew-prior** [comment]
I know that this issue has been closed but I just encountered the same bug. 

I have the line written as `req.onload = function () {` which failed the test. When I wrote it as `req.onload=function(){` it passed the test;

**Comment by QuincyLarson** [comment]
@andrew-prior Thanks for confirming this! I'm reopening the issue so we can further tighten up the tests :)

**Comment by charlottetan** [comment]
Challenge url seems to have moved to
https://learn.freecodecamp.org/data-visualization/json-apis-and-ajax/get-json-with-the-javascript-xmlhttprequest-method/

Additionally, I'm no longer able to repro this. The code snippet posted above completes the challenge successfully.
